tokenserver/scripts/process_account_deletions.py

```diff
@@ -63,21 +63,33 @@ def process_account_deletions(config_file, queue_name, aws_region=None,
             if msg is None:
                 continue
             email = None
+            event_type = None
+            generation = None
             # Try very hard not to error out if there's junk in the queue.
             try:
                 body = json.loads(msg.get_body())
                 event = json.loads(body['Message'])
-                # The queue may contain other event types; ignore them.
-                if event["event"] == "delete":
-                    email = event["uid"]
+                event_type = event["event"]
+                email = event["uid"]
+                if event_type == "reset":
+                    generation = event["generation"]
             except (ValueError, KeyError), e:
-                logger.exception("Invalid account-delete message: %s", e)
+                logger.exception("Invalid account message: %s", e)
             else:
-                # Mark the user as retired.
-                # Actual cleanup is done by a separate process.
                 if email is not None:
-                    logger.info("Processing account deletion for %r", email)
-                    backend.retire_user(email)
+                    if event_type == "delete":
+                        # Mark the user as retired.
+                        # Actual cleanup is done by a separate process.
+                        logger.info("Processing account delete for %r", email)
+                        backend.retire_user(email)
+                    elif event_type == "reset":
+                        logger.info("Processing account reset for %r", email)
+                        user = backend.get_user("sync-1.5", email)
+                        if generation > user['generation']:
```

Comment:
This doesn't need to be conditional, the backend will silently ignore any requests to set the generation number to something smaller than it already is.  Given that, I don't think we even need to worry about additional tests for this feature.